docs/extending/code-overview.rst

```diff
@@ -2,16 +2,22 @@
 Mathics Core Interpreter Overview
 =================================
 
-At its heart, the Mathics interpreter works over `M-expresssion
-<https://en.wikipedia.org/wiki/S-expression>`_: combinations of nested
-and/or sequenced lists, pretty much the same as simple Lisp
-interpreters work. See also `Everything is an Expression
+.. index:: S-expression
+
+At its heart, the Mathics interpreter works over a generalization of
+an `S-expression <https://en.wikipedia.org/wiki/S-expression>`_,
+called an `M-expresssion
+<https://en.wikipedia.org/wiki/M-expression>`_. An M-expression is a
+nested and/or sequenced lists, pretty much the same as simple Lisp
+interpreters work, except the head element can be an expression rather
+than just a simple function name. See also `Everything is an
+Expression
 <https://reference.wolfram.com/language/tutorial/Expressions.html#4715>`_.
 
 When you enter a string to Mathics there is a 3-step process:
 
 * the input string is parsed to ``FullForm``
-* The ``FullForm`` S-expression is evaluated giving another S-expression
+* The ``FullForm`` M-expression is evaluated giving another M-expression
```

Comment:
Weren't we going with S-Expression?

Reply:
@mmatera correctly points out that the way things are implemented what is seen in some cases are M-expressions. 

In particular when you enter `Derivative[1][f]`, `Derivitave[1]` acts like a single function name. 

So we are describing things as the appear in the code.  

A downside of of not using S-expressions which is the more commonly used kind of thing is that most of the literature and work done on lisp-like and functional-like evaluators assumes this. So when there is some technology that comes up and we want to use, we'll have to look this over to see if it is amenable to M-expressions. 

I am pretty sure there is a way at the parser to elminite this so that we do in fact S-expressions, but this isn't something we want to tackle right now.

Reply:
@rocky, as I told you, apart from the comments, I was OK with your changes. If you think that calling *S-Expression* to something that is actually *M-Expression*, for the sake of readability, is OK for me.  Later we can add a footnote.